### PR TITLE
Allow one to rescue multiple exceptions

### PR DESCRIPTION
**Pull request by sshaw**
```rb
File.read(path) rescue [Errno::ENOENT, Errno::EPERM].with { warn "Error: #$!" }
```

### CODE REVIEW

**Comment by codesnik** [comment]
Unlike original code which adds a method to related class, this PR add's exception-rescuing method with a very generic name to an Array. This is the kind of absolutely unnecessary magic and pollution from which even RSpec guys are moving on.

**Comment by sshaw** [comment]
> Unlike original code which adds a method to related class this PR add's exception-rescuing method with a very generic name to an Array
> This is the kind of absolutely unnecessary magic ...

Your argument is drawing a very fine line to justify one use of monkey patching over the other but yes, monkey patching is evil. I get it. We get it. RSpec guys get it (allegedly). This is just for fun.

**Comment by sshaw** [comment]
Alternate method: 6ea653127f0ccdcfb274aed33f6c0b5ad1524cc8. 

Actually I think this is the best 😄

File: lib/rescue-with.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
+class Array
+  def with
+    any? { |e| e === $! } ? yield($!) : fail
+  end
+end
+
```
Comment (sshaw) [inline lib/rescue-with.rb:6]: See new, preferred approach: 6ea6531.

**Comment by ktchen14** [comment]
@sshaw Yeah that's just a little too much monkey patching and if you need to rescue multiple exceptions it's much better to use the `begin ... rescue ... end` syntax. I'll add a note in the README to that effect.

**Pull request closed without merging**